nexus/lib/nexus/peer.ex

```diff
@@ -5,9 +5,45 @@ defmodule Nexus.Peer do
 
   require Logger
 
-  alias ExWebRTC.{ICECandidate, MediaStreamTrack, PeerConnection, RTPCodecParameters}
-  alias Nexus.Chamber
+  alias ExWebRTC.{
+    ICECandidate,
+    MediaStreamTrack,
+    PeerConnection,
+    RTPCodecParameters,
+    SessionDescription
+  }
+
+  alias Nexus.Room
   alias Nexus.Peer.State
+  alias NexusWeb.PeerChannel
+
+  defmodule State do
```

Comment:
I feel like this could be just a `@type` instead of a struct? It's not used anywhere outside of the `Peer` module and I did not notice you using the `%State{state | ...}` syntax much (which would ensure you only use valid fields) so I don't really see benefit of it being a struct. Or maybe the `put_in` from `Bunch.Access` validates the used fields in compile time?

Reply:
Although that's a nitpick so I guess you can leave it as it is.